### ISSUE TITLE
Cloning or adding a revision from existing template has errors sometimes

### ISSUE THREAD

**Issue by soumyalj** [rancher/rancher#23404]
Version: v2.3.1-rc1

Steps:

1. Create a template T1 with revision R1 . Set the below in the template revision.[They are NOT set as overrides]

K8s version - v1.14.7-rancher 1-1
Node port range - 30000-32767
Docker Root Directory - /var/lib/docker123 

2. Create another revision R2 and set the below parameters as questions :

K8s version - v1.15.4-rancher 1-2
Node port range - 30000-32767
Docker Root Directory - /var/lib/docker123 

The console has the below error:

<img width="598" alt="Screen Shot 2019-10-11 at 10 44 01 PM" src="https://user-images.githubusercontent.com/18536626/66695617-aaf62700-ec78-11e9-8de9-7e8dacbe8109.png">

The revision R2 created has the questions field as NULL in the API
```
"ownerReferences": [
{
"apiVersion": "management.cattle.io/v3",
"kind": "ClusterTemplate",
"name": "ct-bnqnx",
"type": "/v3/schemas/ownerReference",
"uid": "909422ab-eca4-11e9-8aba-0242ac110002"
}
],
"questions": null,
"type": "clusterTemplateRevision",
```

This issue is not reproducible all the time. 

Note: Since the template revision got created without questions,when using this revision during cluster edit, UI will not display the fields as questions.

**Comment by westlywright** [comment]
I could not reproduce this with the steps given, but the error did provide enough info that I've added some code to ensure where its blowing up in the code above shouldn't happen again.

**Comment by westlywright** [comment]
Available in UI tag `2.3.17`

**Comment by soumyalj** [comment]
Tested with rancher v2.3.1-rc2 with UI tag `2.3.17`

1. Create a template T1 with revision R1 . Set the below in the template revision.[They are NOT set as overrides]
K8s version - v1.14.7-rancher 1-1
Node port range - 30000-32767
Docker Root Directory - /var/lib/docker123

2. Create another revision R2 and set the below parameters as questions :
K8s version - v1.15.4-rancher 1-2
Node port range - 30000-32767
Docker Root Directory - /var/lib/docker123

R2 got created correctly with questions as below:
```
"questions": [
{
"default": "v1.15.4-rancher1-2",
"max": 0,
"maxLength": 0,
"min": 0,
"minLength": 0,
"required": false,
"type": "string",
"variable": "rancherKubernetesEngineConfig.kubernetesVersion"
}
],
```

Also tried the use case below:
R1 is created with K8s version as override. 
R2 is created as a clone and the UI presents the K8s version field as an override correctly. The API shows the questions correctly

The use cases were repeated 5 times and errors as mentioned in the issue were not observed.
